packages/devextreme/js/ui/date_box/ui.date_box.mask.js

```diff
@@ -331,6 +336,18 @@ const DateBoxMask = DateBoxBase.inherit({
         }
     },
 
+    _handleAmPmSearch(char) {
+        const startString = this._searchValue + char.toLowerCase();
+
+        this._searchValue = startString;
+
+        if(char.toLowerCase() === 'p' && this._getActivePartProp('text') === 'AM') {
+            this._toggleAmPm();
+        } else if(char.toLowerCase() === 'a' && this._getActivePartProp('text') === 'PM') {
+            this._toggleAmPm();
+        }
```

Comment:
```suggestion
if ((char === 'p' && activePartText === 'am') || (char === 'a' && activePartText === 'pm')) {
    this._toggleAmPm();
}
```
activePartText can be passed as a function parameter, plz look at my commets above.